app/controllers/users_controller.rb

```diff
@@ -22,4 +22,20 @@ def contributions
       per_page: per_page
     ).build_hash, status: :ok
   end
+
+  def reviews
+    per_page = params[:per_page] || 5
+    page = params[:page] || 1
+
+    render json: PaginatedSerializableService.new(
+      records: current_user .reviews,
```

Comment:
It's not really "reviews". It's "reviewed gestures".

The structure of the response of `reviewed_gestures` and `contributions` should be exactly the same, as opposed to what you're doing. This endpoint should NOT return reviews primarily and then gesture, gesture.word in relations. It should return the gesture primarily, same as the contributions endpoint for consistency or symmetry. Let me know if you disagree.

What I would suggest you do is to create an activerecord relation between our Reviewer (User subclass) class and Gesture class. The Reviewer should have many Gestures through Review.

Rails has a [specific relation type](https://guides.rubyonrails.org/association_basics.html#the-has-many-through-association) for this, called the has-many-through association. You should call the relation reviewed_gestures, so that it'll be callable as for ex. `current_user.reviewed_gestures` otherwise it'll clash with the gestures relation we already have setup in the User class.

After you setup the relation. The endpoint will be almost identical to contributions.

Reply:
I think the endpoint is primarily meant to return a user's reviews (the review date, comment, response). I don't see why this has to be the same structure as contributions.